invenio_records_permissions/policies/base.py

```diff
@@ -114,6 +115,20 @@ def excludes(self):
         self._load_permissions()  # self.explicit_excludes is used here
         return self._permissions.excludes
 
+    def _query_filters_superuser(self, filters):
+        """Allow superuser identity to retrieve all results."""
+        identity = self.over.get("identity")
+        identity_provides = identity.provides if identity else set()
+        # expand action to resolve needs
+        expanded = self._expand_action(superuser_access)
+        superuser_needs = expanded.needs if expanded else set()
+
+        is_superuser = superuser_needs & identity_provides
+        if is_superuser:
+            filters.append(Q("match_all"))
```

Comment:
note for self: Here we append a `match_all`. Since the generator does an `|` (OR) between filters, it will translate into a `should` (not `must`) in the DSL. So it's correct.